front/app/containers/Admin/messaging/texting/CampaignList/TextCampaignListRow.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+
+// routing
+import { withRouter, WithRouterProps } from 'react-router';
+import Link from 'utils/cl-router/Link';
+
+// components
+import { StatusLabel } from '@citizenlab/cl2-component-library';
+import messages from '../../messages';
+import Button from 'components/UI/Button';
+
+// typings
+import { ITextingCampaignData } from 'services/textingCampaigns';
+
+// style
+import styled from 'styled-components';
+import { colors } from 'utils/styleUtils';
+
+// i18n
+import { FormattedTime, FormattedDate } from 'react-intl';
+import { FormattedMessage } from 'utils/cl-intl';
+
+interface Props {
+  campaign: ITextingCampaignData;
+}
+
+const Container = styled.div`
+  width: 100%;
+  height: 50px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  border-top: 1px solid #e0e0e0;
+`;
+
+const Left = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 20px;
+  width: 100%;
+`;
+
+const TextWrapper = styled.div`
+  line-height: 50px;
+`;
+
+const Text = styled.p`
+  width: 600px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  font-size: 16px;
+`;
+
+const Right = styled.div`
+  display: flex;
+  gap: 30px;
+`;
+
+// 140px is sufficient for longest date-time string
+const DateTime = styled.div`
+  width: 140px;
+`;
+
+// 174px is sufficient for 999,999 recipients
+const StatusWrapper = styled.div`
+  width: 174px;
+  text-align: right;
+`;
+
+const TextingCampaignRow = ({ campaign }: Props & WithRouterProps) => {
```

Comment:
WithRouter is not being used in this component, so not sure why it's added (guess copy-paste?). Removing it will save a nice import (15K gzipped).

Reply:
Ahh, I thought it was necessary for `Link` component but that seems to be incorrect. Will remove.